code/clue.js

```diff
@@ -1,59 +1,347 @@
+  
+  weapons = [
+    rope, 
+    candlestick, 
+    revolver, 
+    leadPipe, 
+    dagger, 
+    wrench
+  ];
+  weapons.forEach(findFavouriteWeapon);
+
+  //Styling
+  document.getElementById('killerInner').style.background = mystery.killer.color;
+  document.getElementById('killerCard').classList.remove('active');
+  document.getElementById('killerInner').classList.remove('front');
+  document.getElementById('killerInner').classList.add('back');
+
+  //Show information
   document.getElementById(
     'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  document.getElementById('killerAge').innerHTML = `${mystery.killer.age}`;
+  document.getElementById('killerOccupation').innerHTML = `${mystery.killer.occupation}`;
+  document.getElementById('killerImage').src = `${mystery.killer.image}`;
+  document.getElementById(
+    'killerDescription').innerHTML = `${mystery.killer.description}`;
+  document.getElementById('killerFavWeapon').innerHTML = `Favourite weapon: ${mystery.killer.favouriteWeapon}`;
+};
+
+
+// 2. for the weapon card deck.
+
+const pickWeapon = () => {
+  document.getElementById('weaponLoader').style.display = 'none';
+
+  mystery.weapon = randomSelector(weapons);
+
+  rooms = [
+    kitchen,
+    ballroom,
+    conservatory,
+    billiardRoom,
+    library,
+    study,
+    hall,
+    lounge,
+    diningRoom
+  ];
+  rooms.push(
+    mystery.weapon.favouriteRoom,
+    mystery.weapon.favouriteRoom,
+    mystery.weapon.favouriteRoom,
+    mystery.weapon.favouriteRoom,
+  );
+  rooms.push(
+    mystery.weapon.secondFavouriteRoom,
+    mystery.weapon.secondFavouriteRoom,
+  );
+
+  //Styling
+  document.getElementById('weaponCard').classList.remove('active');
+  document.getElementById('weaponInner').classList.remove('front');
+  document.getElementById('weaponInner').classList.add('back');
+
+  //Show information
+  document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`;
+  document.getElementById('weaponWeight').innerHTML = `${mystery.weapon.weight}`;
+  document.getElementById('weaponMaterial').innerHTML = `${mystery.weapon.material}`;
+};
+
+
+//3. for the room card deck.
+
+const pickRoom = () => {
+  document.getElementById('roomLoader').style.display = 'none';
+
+  mystery.room = randomSelector(rooms);
+
+  //Styling
+  document.getElementById('roomCard').classList.remove('active');
+  document.getElementById('roomInner').classList.remove('front');
+  document.getElementById('roomInner').classList.add('back');
+
+  //Show information
+  document.getElementById('roomName').innerHTML = `${mystery.room}`;
+};
+
+
+// REAVEALING THE MYSTERY
+
+const revealMystery = () => {
+  const revealText = document.getElementById('revealedMystery');
+  revealText.style.display = 'block';
+
+  if (mystery.killer !== null && mystery.weapon !== null && mystery.room !== null) {
```

Comment:
Or we could simplify this statement like this :

```
if (mystery.killer && mystery.weapon && mystery.room) { ... }
```

or eventually 

```
if (!!mystery.killer && !!mystery.weapon && !!mystery.room) { ... }
```

For further informations please check https://www.freecodecamp.org/news/js-type-coercion-explained-27ba3d9a2839/